pkg/raft/log_test.go

```diff
@@ -674,6 +721,14 @@ func TestLogRestore(t *testing.T) {
 	require.Equal(t, index, raftLog.committed)
 	require.Equal(t, index, raftLog.unstable.prev.index)
 	require.Equal(t, term, mustTerm(raftLog.term(index)))
+	// term cache should be re-initialized to only have the snapshot
+	// meta entryID.
+	require.Equal(t, term, raftLog.termCache.first().term)
+	require.Equal(t, 1, len(raftLog.termCache.cache))
+	// term cache should return false for term lookup for index less
+	// than snapshot index after restoring snapshot.
+	term, found := raftLog.termCache.term(index - 1)
+	require.False(t, found)
```

Comment:
```suggestion
```
This check is unnecessary. The above checks that `first() == id` and `len == 1` are sufficient, and imply that < `id.index` return not found.